lib/ipsecbase.pm

```diff
@@ -0,0 +1,120 @@
+# SUSE's openQA tests
+#
+# Copyright 2016-2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Summary: Base module for IPsec tests
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+package ipsecbase;
+use Mojo::Base 'opensusebasetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use Utils::Architectures;
+use version_utils 'is_sle';
+use lockapi;
+use Utils::Logging 'save_and_upload_log';
+use network_utils;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = $class->SUPER::new($args);
+
+    $self->{ipsec_id} = "0x26c44388";
+    $self->{ipsec_key} = "0x6f887514ca6eccb1d273366f70b21a91aa2a3421";
+    $self->{ipsec_mode} = "tunnel";
+    $self->{ipsec_aead} = "'rfc4106(gcm(aes))'";
+    $self->{ipsec_proto} = "esp";
+
+    return $self;
+}
+
+sub get_net_prefix_len {
+    my ($self, %args) = @_;
+    my ($len) = $args{net} =~ /\/(\d+)/;
+    return $len;
+}
+
+sub add_ipv6_addr {
+    my ($self, %args) = @_;
+    $args{dev} ||= iface();
+    $args{plen} ||= 64;
+    $args{ip} = $args{ip} . "/" . $args{plen};
+    assert_script_run("ip -6 addr add $args{ip} dev $args{dev}");
+}
+
+sub add_ipv6_route {
+    my ($self, %args) = @_;
+    assert_script_run("ip -6 route add $args{dst} via $args{via}");
+}
+
+sub upload_service_log {
+    my ($self, $service_name, $args) = @_;
+    script_run("journalctl -u $service_name -o short-precise > /tmp/$service_name");
+    script_run("cat /tmp/$service_name");
+    upload_logs("/tmp/$service_name", failok => 1, log_name => $args->{log_name});
+}
+
+sub destroy_test_barriers {
+    my ($self) = @_;
+    barrier_destroy('IPSEC_IP_SETUP_DONE');
+    barrier_destroy('IPSEC_ROUTE_SETUP_DONE');
+    barrier_destroy('IPSEC_TUNNEL_MODE_SETUP_DONE');
+    barrier_destroy('IPSEC_SET_MTU_DONE');
+    barrier_destroy('IPSEC_TUNNEL_MODE_CHECK_DONE');
+    barrier_destroy('IPSEC_TRANSPORT_MODE_SETUP_DONE');
+    barrier_destroy('IPSEC_TRANSPORT_MODE_CHECK_DONE');
+}
+
+sub post_run_hook {
+    my ($self) = @_;
+    select_console('log-console');
+    my $hname = get_var('HOSTNAME', 'susetest');
+    $self->upload_service_log("wicked");
+    upload_logs('/var/log/zypper.log');
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    #$self->destroy_test_barriers();
+    select_serial_terminal;
+    script_run("SUSEConnect --status-text");
+    script_run("journalctl -o short-precise > /tmp/journal.log");
+    script_run('cat /tmp/journal.log');
+    upload_logs('/tmp/journal.log', failok => 1);
+    upload_service_log('wickedd-dhcp4.service');
```

Comment:
Aren't many of of these  things part of default post fail hook? If you need additional logs, just hook via self and add additional later